packages/graphqlsp/src/diagnostics.ts

```diff
@@ -247,6 +250,26 @@ export function getGraphQLDiagnostics(
           };
         }
 
+        const hash = callExpression.arguments[0].getText().slice(1, -1);
+        if (hash.startsWith('sha256:')) {
+          const upToDateHash = `sha256:${generateHashForDocument(
+            info,
+            initializer.arguments[0],
+            foundFilename
+          )}`;
+          if (upToDateHash !== hash) {
+            return {
+              category: ts.DiagnosticCategory.Warning,
+              code: MISSMATCH_HASH_TO_DOCUMENT,
+              file: source,
+              messageText: `The Persisted-Document hash is not up to date with the linked document.`,
```

Comment:
I'd negate "up-to-date"and drop the redundant dash ✌️

```suggestion
              messageText: `The persisted document's hash is outdated`,
```